operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -98,6 +99,10 @@ public class DomainProcessorImpl implements DomainProcessor, MakeRightExecutor {
   @SuppressWarnings({"FieldMayBeFinal", "CanBeFinal"})
   private static Map<String, Map<String, DomainPresenceInfo>> domains = new ConcurrentHashMap<>();
 
+  // Map namespace to map of clusterName to ClusterResource; tests may replace this value.
+  @SuppressWarnings({"FieldMayBeFinal", "CanBeFinal"})
+  private static Map<String, Map<String, ClusterResource>> clusters = new ConcurrentHashMap<>();
```

Comment:
As part of DomainValidationSteps I believe we call createListClustersStep to return a list of Cluster resources in the same namespace. What is the reason that we are not able to use that list of Cluster resources for the new validation is being added in this PR, instead of creating a new list of ClusterResource here? I may be missing something obvious though. Thanks.

Reply:
Yes, DomainValidationSteps uses createListClustersStep to get all clusters, and it puts the list into the packet. I thought about using the list in the packet before I added the new map. But at the time, somehow I got the impression that the list in the packet only contains the clusters that are referenced by the the domain that the operator is processing. Not I read the code again, and I believe that the list has all clusters in the namespace. I am glad that we don't need to use a new map after all. I'll give it a try and I think it should work.  thanks for catching this.